bson/src/main/org/bson/types/Decimal128.java

```diff
@@ -444,11 +504,13 @@ public double doubleValue() {
             }
         }
 
-        if (equals(NEGATIVE_ZERO)) {
-            return 0.0;
+        BigDecimal bigDecimal = bigDecimalValueNoNegativeZeroCheck();
+
+        if (hasDifferentSign(bigDecimal)) {
```

Comment:
Fixed a bug here.  I forgot that with Decimal128 there are many essentially equal values.  NEGATIVE_ZERO is just "-0", but there is also "-0.0", "-0.00", etc, none of which will be equal to NEGATIVE_ZERO.

Fixed by using the same trick as we do in bigDecimalValue to see if the value is some form of negative zero.

Reply:
Ack